### PR TITLE
Add codespell to CI

### LINKED ISSUE #5659: Add codespell to CI
Add `codespell` Make target and GitHub workflow like in https://github.com/open-telemetry/opentelemetry-go. References:
- https://github.com/open-telemetry/opentelemetry-go/blob/main/.github/workflows/codespell.yaml
- https://github.com/open-telemetry/opentelemetry-go/blob/main/Makefile

### PR DESCRIPTION
**Pull request by SequoiaGod**
Closes #5659 

### Summary of the changes in the code

1.  Added the codespell GitHub workflow.
2. Added makefile command which is related the codespell.
3. Added the requirements.txt file which is used for install the dependency package of Python.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -11,7 +11,7 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 ### Added
 
 - The `go.opentelemetry.io/contrib/config` add support to configure periodic reader interval and timeout. (#5661)
-
+- The codespell workflow has been added to GitHub Action, and also populate related makefile command and requirements.txt file. (#5683)
```
Comment (dmathieu) [inline CHANGELOG.md:14]: This change is internal to the repository (it has no impact on the provided libraries). So there's no changelog needed.

Comment (SequoiaGod) [reply CHANGELOG.md:14]: Yup, thanks for pointing it out. I will get rid of this line mate.

File: .codespellignore [hunk 2]
```diff
@@ -0,0 +1,10 @@
+ot
+fo
+te
+collison
+consequentially
+ans
+nam
+valu
```
Comment (dmathieu) [inline .codespellignore:8]: `value`?

File: .codespellignore [hunk 3]
```diff
@@ -0,0 +1,10 @@
+ot
+fo
+te
+collison
```
[the remark this answers is not in the document]
Comment (SequoiaGod) [reply .codespellignore:4]: LOL, I copy some of the words from the codespellignore file of opentelemetry-go project. I used to guess those might have different naming meanings. I will quickly amend those typos.

File: .codespellignore [hunk 4]
```diff
@@ -0,0 +1,10 @@
+ot
+fo
```
[the remark this answers is not in the document]
Comment (SequoiaGod) [reply .codespellignore:2]: It's not of. It is a wildcard in the code `		{"fo*", "boo"},`

File: .codespellignore [hunk 5]
```diff
@@ -0,0 +1,10 @@
+ot
```
[the remark this answers is not in the document]
Comment (SequoiaGod) [reply .codespellignore:1]: Having dived into the code, I am guessing it's not typo of to, as I found this is part of the module name in the code : 
`      - go.opentelemetry.io/contrib/propagators/ot`

File: config/metric.go [hunk 6]
```diff
@@ -222,7 +222,7 @@ func prometheusReader(ctx context.Context, prometheusConfig *Prometheus) (sdkmet
 	mux := http.NewServeMux()
 	mux.Handle("/metrics", promhttp.HandlerFor(reg, promhttp.HandlerOpts{Registry: reg}))
 	server := http.Server{
-		// Timeouts are necessary to make a server resilent to attacks, but ListenAndServe doesn't set any.
+		// Timeouts are necessary to make a server  resilient to attacks, but ListenAndServe doesn't set any.
```
Comment (dmathieu) [inline config/metric.go:225]:
```suggestion
		// Timeouts are necessary to make a server resilient to attacks, but ListenAndServe doesn't set any.
```

File: requirements.txt [hunk 7]
```diff
@@ -0,0 +1 @@
+codespell==2.3.0
```
Comment (dmathieu) [inline requirements.txt:1]: Should we add this to the renovabot upgrades to check for new versions?

File: doc.go [hunk 8]
```diff
@@ -2,6 +2,6 @@
 // SPDX-License-Identifier: Apache-2.0
 
 // Package contrib is a collection of extensions for the opentelemetry-go
-// project. It provides 3rd parth resource detectors, propagators, samplers,
+// project. It provides 3rd path resource detectors, propagators, samplers,
```
Comment (MrAlias) [inline doc.go:5]:
```suggestion
// project. It provides 3rd party resource detectors, propagators, samplers,
```

File: requirements.txt (hunk 7 above)
Comment (SequoiaGod) [reply requirements.txt:1]: added to the renovate.json

**Review by dmathieu — APPROVED** [review]

**Review by MrAlias — APPROVED** [review]

**Pull request merged**